pennylane_lightning/core/src/simulators/lightning_gpu/utils/cuStateVecError.hpp

```diff
@@ -0,0 +1,110 @@
+// and from Lightning: https://github.com/PennylaneAI/pennylane-lightning.git
+
+/**
+ * @file cuStateVecError.hpp
+ * Defines macros that throws Exception from cuStateVec failure error codes.
+ */
+
+#pragma once
+#include <string>
+
+#include <custatevec.h>
+
+#include "Error.hpp"
+#include "Util.hpp"
+
+// LCOV_EXCL_START
+/// @cond DEV
+namespace {
+using namespace Pennylane::Util;
+}
+/// @endcond
+
+#ifndef CUDA_UNSAFE
+
+/**
+ * @brief Macro that throws Exception from cuStateVec failure error codes.
+ *
+ * @param err cuStateVec function error-code.
+ */
+#define PL_CUSTATEVEC_IS_SUCCESS(err)                                          \
+    PL_ABORT_IF_NOT(                                                           \
+        err == CUSTATEVEC_STATUS_SUCCESS,                                      \
+        Pennylane::LightningGPU::Util::GetCuStateVecErrorString(err).c_str())
+
+#else
+#define PL_CUSTATEVEC_IS_SUCCESS(err)                                          \
+    { static_cast<void>(err); }
+#endif
+
+namespace Pennylane::LightningGPU::Util {
+static const std::string
+GetCuStateVecErrorString(const custatevecStatus_t &err) {
+    std::string result;
+    switch (err) {
+    case CUSTATEVEC_STATUS_SUCCESS:
+        result = "No errors";
+        break;
+    case CUSTATEVEC_STATUS_NOT_INITIALIZED:
+        result = "custatevec not initialized";
+        break;
+    case CUSTATEVEC_STATUS_ALLOC_FAILED:
+        result = "custatevec memory allocation failed";
+        break;
+    case CUSTATEVEC_STATUS_INVALID_VALUE:
+        result = "custatevec invalid value";
+        break;
+    case CUSTATEVEC_STATUS_ARCH_MISMATCH:
+        result = "custatevec CUDA device architecture mismatch";
+        break;
+    case CUSTATEVEC_STATUS_EXECUTION_FAILED:
+        result = "custatevec execution failed";
+        break;
+    case CUSTATEVEC_STATUS_INTERNAL_ERROR:
+        result = "custatevec internal error";
+        break;
+    case CUSTATEVEC_STATUS_NOT_SUPPORTED:
+        result = "custatevec unsupported operation/device";
+        break;
+    case CUSTATEVEC_STATUS_INSUFFICIENT_WORKSPACE:
+        result =
+            "custatevec insufficient memory for gate-application workspace";
+        break;
+    case CUSTATEVEC_STATUS_SAMPLER_NOT_PREPROCESSED:
+        result = "custatevec sampler not preprocessed";
+        break;
+    case CUSTATEVEC_STATUS_NO_DEVICE_ALLOCATOR:
+        result = "custatevec no device allocator";
+        break;
+    case CUSTATEVEC_STATUS_DEVICE_ALLOCATOR_ERROR:
+        result = "custatevec device allocator error";
+        break;
+    case CUSTATEVEC_STATUS_COMMUNICATOR_ERROR:
+        result = "custatevec communicator failure";
+        break;
+    case CUSTATEVEC_STATUS_LOADING_LIBRARY_FAILED:
+        result = "custatevec dynamic library load failure";
+        break;
+    default:
+        result =
+            "custatevec status not found. Error code=" + std::to_string(err);
+    }
+    return result;
+}
```

Comment:
Please consider using string_views's literals bindings to simplify the code and avoid unnecessary std::string construction.  
```suggestion
    using namespace std::string_literals;

    switch (err) {
    case CUSTATEVEC_STATUS_SUCCESS:
        return "No errors"s;
    case CUSTATEVEC_STATUS_NOT_INITIALIZED:
        return "custatevec not initialized"s;
    case CUSTATEVEC_STATUS_ALLOC_FAILED:
        return "custatevec memory allocation failed"s;
    case CUSTATEVEC_STATUS_INVALID_VALUE:
        return "custatevec invalid value"s;
    case CUSTATEVEC_STATUS_ARCH_MISMATCH:
        return "custatevec CUDA device architecture mismatch"s;
    case CUSTATEVEC_STATUS_EXECUTION_FAILED:
        return "custatevec execution failed"s;
    case CUSTATEVEC_STATUS_INTERNAL_ERROR:
        return "custatevec internal error"s;
    case CUSTATEVEC_STATUS_NOT_SUPPORTED:
        return "custatevec unsupported operation/device"s;
    case CUSTATEVEC_STATUS_INSUFFICIENT_WORKSPACE:
        return "custatevec insufficient memory for gate-application workspace"s;
    case CUSTATEVEC_STATUS_SAMPLER_NOT_PREPROCESSED:
        return "custatevec sampler not preprocessed"s;
    case CUSTATEVEC_STATUS_NO_DEVICE_ALLOCATOR:
        return "custatevec no device allocator"s;
    case CUSTATEVEC_STATUS_DEVICE_ALLOCATOR_ERROR:
        return "custatevec device allocator error"s;
    case CUSTATEVEC_STATUS_COMMUNICATOR_ERROR:
        return "custatevec communicator failure"s;
    case CUSTATEVEC_STATUS_LOADING_LIBRARY_FAILED:
        return "custatevec dynamic library load failure"s;
    default:
        return "custatevec status not found. Error code="s +
               std::to_string(err);
}
```